File: banchus-common/src/main/java/pe/nanamochi/banchus/database/repository/ChannelRepository.java
package pe.nanamochi.banchus.database.repository;

import java.util.List;
import java.util.Optional;
import java.util.UUID;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import pe.nanamochi.banchus.database.entity.Channel;

@Repository
public interface ChannelRepository extends JpaRepository<Channel, UUID> {
  List<Channel> findByAutoJoin(boolean autoJoin);

  Optional<Channel> findByName(String name);
}

File: banchus-common/src/main/kotlin/pe/nanamochi/banchus/database/repository/ChannelRepository.kt
package pe.nanamochi.banchus.database.repository

import java.util.UUID
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import pe.nanamochi.banchus.database.entity.Channel

@Repository
interface ChannelRepository : JpaRepository<Channel, UUID> {
    fun findByName(name: String): Channel?
}
